pkg/kubecost/allocation.go

```diff
@@ -601,9 +602,38 @@ func (a *Allocation) add(that *Allocation) {
 		log.Warningf("Allocation.AggregateBy: trying to add a nil receiver")
 		return
 	}
+
+	// Save the original properties so we can set a controller name in the event of
+	// equality in all fields except the controller name
+	leftProperties := a.Properties
+	rightProperties := that.Properties
+
 	// Preserve string properties that are matching between the two allocations
 	a.Properties = a.Properties.Intersection(that.Properties)
 
+	// Overwrite regular intersection logic for the controller name property in the
+	// case that all properties except controller name are the same.
+	if leftProperties != nil &&
+		rightProperties != nil &&
+		leftProperties.Controller != rightProperties.Controller &&
+		leftProperties.Controller != "" &&
+		rightProperties.Controller != "" {
```

Comment:
I don't know the answer here, but are we certain that we want to do this only when _both_ Controllers are `""`?

Reply:
i.e. when one is empty and the other is not, we're happy with the current logic?

Reply:
That's a good call out. I think it's reasonable to also do this when only one is empty string but we should think about edge cases. I don't have time to do so today, but I can check it out in the morning.

Reply:
(if we do include the empty string case, note to self to update the sort logic so that we don't pick the empty string)